File: EstructuraDeDatos/ClasesBase/CEstudiante.cpp
// CEstudiante.cpp
#include "CEstudiante.h"
Elemento::Elemento()
{
 //ctor
}
Elemento::~Elemento()
{
 //dtor
}
int Elemento::Leer(int ind)
{
 cout << "ALUMNO[" << ind << "], cedula (salir = NULA): ";
 gets(m_Cedula);
 if (strlen(m_Cedula) == 0) return 0;
 cout << "nombre (salir = NULO): ";
 gets(m_Nombre);
 if (strlen(m_Nombre) == 0) return 0;
 cout << "notas de los 3 parciales: ";
 cin >> m_Nota1 >> m_Nota2 >> m_Nota3;
 // Limpiar el buffer:
 getchar();
 return 1;
}
int Elemento::Leer(FILE *F)
{
 if (!fgets(m_Cedula, 50, F)) return 0;
 m_Cedula[strlen(m_Cedula) - 1] = '\0';
 if (!fgets(m_Nombre, 50, F)) return 0;
 m_Nombre[strlen(m_Nombre) - 1] = '\0';
 fscanf(F, "%f %f %f", &m_Nota1, &m_Nota2, &m_Nota3);
 // Limpiar el buffer:
 fgetc(F);
 return 1;
}
void Elemento::Mostrar()
{
 float prom = (m_Nota1 + m_Nota2 + m_Nota3) / 3;
 printf("%s %s %.2f %.2f %.2f %.2f %s\n",
 m_Cedula, m_Nombre, m_Nota1, m_Nota2, m_Nota3, prom,
 prom < 14 ? "REPRUEBA" : "APRUEBA");
}
void Elemento::Mostrar(FILE *G)
{
 float prom = (m_Nota1 + m_Nota2 + m_Nota3) / 3;
 fprintf(G, "%s %s %.2f %.2f %.2f %.2f %s\n",
 m_Cedula, m_Nombre, m_Nota1, m_Nota2, m_Nota3, prom,
 prom < 14 ? "REPRUEBA" : "APRUEBA");
}
int Elemento::Compara(Elemento x)
{
 if (m_Indicador == 0)
 return strcmpi(m_Cedula, x.GetCedula());
 return strcmpi(m_Nombre, x.GetNombre());
}
void Elemento::Copiar(Elemento x)
{
 *this = x;
}

File: EstructuraDeDatos/Python/ModulosPY/CEstudiante.py
# CEstudiante.py
class CEstudiante:
    Indicador = 0
    def __init__(elm, ced, nom, nt1, nt2, nt3):
        elm.cedula = ced
        elm.nombre = nom
        elm.nota1 = nt1
        elm.nota2 = nt2
        elm.nota3 = nt3

    def Leer(elm, ind):
        cad = input("ALUMNO[" + str(ind) + "], cedula (salir = NULA): ")
        if len(cad) == 0:
            return elm, 0
        elm.cedula = cad
        cad = input("nombre (salir = NULO): ")
        if len(cad) == 0:
            return elm, 0
        elm.nombre = cad
        elm.nota1 = float(input("nota 1: "))
        elm.nota2 = float(input("nota 2: "))
        elm.nota3 = float(input("nota 3: "))
        return elm, 1
    def Mostrar(x):
        prom = round((x.nota1 + x.nota2 + x.nota3) / 3, 2)
        print(x.cedula, " ", x.nombre, " ", x.nota1, " ", x.nota2, " ", x.nota3, " ",
        prom, "REPRUEBA" if prom < 14 else "APRUEBA")
    def Comparar(elm, y):
        if CEstudiante.Indicador == 0:
            if elm.cedula < y.cedula:
                return -1
            if elm.cedula > y.cedula:
                return 1
            return 0
        if elm.nombre < y.nombre:
            return -1
        if elm.nombre > y.nombre:
            return 1
        return 0
    def SetCedula(elm, x):
        elm.cedula = x